src/ExadelMentorship.BusinessLogic/CommandInvoker.cs

```diff
@@ -0,0 +1,31 @@
+﻿using ExadelMentorship.BusinessLogic.Exceptions;
+using ExadelMentorship.BusinessLogic.Features;
+using ExadelMentorship.BusinessLogic.Features.WeatherFeature;
+using ExadelMentorship.BusinessLogic.Interfaces;
+using ExadelMentorship.BusinessLogic.Models;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ExadelMentorship.BusinessLogic
+{
+    public class CommandInvoker
+    {
+        private readonly IServiceProvider _serviceProvider;
+        public CommandInvoker(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public Task Invoke(ICommand command) 
+        {
+            var handler = _serviceProvider.GetRequiredService<ICommandHandler<ICommand>>();
```

Comment:
You can't resolve `ICommandHandler<ICommand>` services, as they will registered with concrete classes i.e. `ICommandHandler<concrete-type-of-ICommand>`. So it requires using `dynamic` technique.

1. Create private method `private Task Invoke<T>(T Command) where T : ICommand`. It will be a simple wrapper/helper method for type inference
2. Call that private method from here simply with `return Invoke((dynamic) command)`;

Reply:
done, please review